contrib/platform/src/com/sun/jna/platform/win32/WbemcliUtil.java

```diff
@@ -0,0 +1,682 @@
+        BSTR namespaceStr = OleAuto.INSTANCE.SysAllocString(namespace);
+        HRESULT hres = loc.ConnectServer(namespaceStr, null, null, null, 0, null, null, pSvc);
+        OleAuto.INSTANCE.SysFreeString(namespaceStr);
+        // Release the locator. If successful, pSvc contains connection
+        // information
+        loc.Release();
+        if (COMUtils.FAILED(hres)) {
+            throw new WbemcliException(String.format("Could not connect to namespace %s.", namespace), hres.intValue());
+        }
+
+        // Step 5: --------------------------------------------------
+        // Set security levels on the proxy -------------------------
+        hres = Ole32.INSTANCE.CoSetProxyBlanket(pSvc.getValue(), Ole32.RPC_C_AUTHN_WINNT, Ole32.RPC_C_AUTHZ_NONE, null,
+                Ole32.RPC_C_AUTHN_LEVEL_CALL, Ole32.RPC_C_IMP_LEVEL_IMPERSONATE, null, Ole32.EOAC_NONE);
+        if (COMUtils.FAILED(hres)) {
+            new IWbemServices(pSvc.getValue()).Release();
+            throw new WbemcliException("Could not set proxy blanket.", hres.intValue());
+        }
+        return new IWbemServices(pSvc.getValue());
+    }
+
+    /**
+     * Selects properties from WMI. Returns immediately (asynchronously), even
+     * while results are being retrieved; results may begun to be enumerated in
+     * the forward direction only.
+     *
+     * @param svc
+     *            A WbemServices object to make the calls
+     * @param query
+     *            A WmiQuery object encapsulating the details of the query
+     * @return An enumerator to receive the results of the query
+     */
+    public static <T extends Enum<T>> IEnumWbemClassObject selectProperties(IWbemServices svc, WmiQuery<T> query) {
+        PointerByReference pEnumerator = new PointerByReference();
+        // Step 6: --------------------------------------------------
+        // Use the IWbemServices pointer to make requests of WMI ----
+        T[] props = query.getPropertyEnum().getEnumConstants();
+        StringBuilder sb = new StringBuilder("SELECT ");
+        // We earlier checked for at least one enum constant
+        sb.append(props[0].name());
+        for (int i = 1; i < props.length; i++) {
+            sb.append(',').append(props[i].name());
+        }
+        sb.append(" FROM ").append(query.getWmiClassName());
+        // Send the query. The flags allow us to return immediately and begin
+        // enumerating in the forward direction as results come in.
+        BSTR queryStr = OleAuto.INSTANCE.SysAllocString(sb.toString().replaceAll("\\\\", "\\\\\\\\"));
+        BSTR wql = OleAuto.INSTANCE.SysAllocString("WQL");
+        HRESULT hres = svc.ExecQuery(wql, queryStr,
+                Wbemcli.WBEM_FLAG_FORWARD_ONLY | Wbemcli.WBEM_FLAG_RETURN_IMMEDIATELY, null, pEnumerator);
+        OleAuto.INSTANCE.SysFreeString(queryStr);
+        OleAuto.INSTANCE.SysFreeString(wql);
+        if (COMUtils.FAILED(hres)) {
+            svc.Release();
```

Comment:
It was released there in the [sample code that I ported](https://docs.microsoft.com/en-us/windows/desktop/WmiSdk/example--getting-wmi-data-from-the-local-computer), although since I've moved out other resource releasing, may as well do that here, as well.